PR Review + Knock Sounds

hanss314 opened the pull request:
Fixed some things in the last PR and added non-meme knock sounds

hanss314 committed:
Completed PR Review 2

Knock particles appear at fist
Set transition durations to 0
Difficulties changed
    St. 2: door moves
    St. 3: door teleports

Fixed cursor hitbox (this was an oversight on my part previously)
Door doesn't do appear animation on start
Moved win animation to another layer
Disabled knocking after victory

hanss314 committed:
Update Knock Sound

Gman8r reviewed (changes requested):
- The knock particles play when you start the scene. Uncheck the Play on Awake option. I'm gonna take a stab and say you probably didn't notice this because you weren't using R to restart the scene, so make sure you do that and make sure the game behaves properly when it restarts
- The particles also move with the hand when logically they shouldn'y and should just stay where they are after they're created. Change simulation space to World to fix that
- The hand is still way above the cursor itself. Pause and unpause during the game to see where I mean. Just lower the hand child object.
- I didn't notice this the first time but you're going to have to fix the door prefab. Look over your scene hierachies again. Your first scene only uses a child prefab under Door called "Rig" and Door itself is unprefabed, and none of the other scenes have the door as a prefab. There's also a duplicate Door prefab asset in the same folder as Rig. Drag the Door object from the first scene into the prefab called "Door" to save it as that prefab, then delete "Rig" because we're not using it, then apply that door prefab to the rest of the scenes, replacing the Door in them.
- The second scene door doesn't need to change direction when it's knocked on, It doesn't really add much. Just make the door slightly faster and have the knock have no bearing on its direction.

hanss314 committed:
Complete requested changes

Knock particles don't play on start
Particle simulation space changed to World
Door doesn't change direction when knocked
Correct fist position to be on the cursor
Fix door prefabs
- Remove unused Rig prefab
- Link all Door prefabs
- Create stage specific Okina prefabs
-- Different Okina child objects would cause prefabs to break
-- This is also so I don't lose the Okina child objects

Gman8r commented:
Okidokes goodo, let's lay out some changes for next time though:

- Let's put a small (and adjustable via inspector) cooldown timer on the hand before you can knock again. Nothing huge, but something to prevent the game from feeling like just a half second of jitterclicking.
- But on that topic, it looks like you have two separate mouse/collider checks for the knock action: one in the Hand for playing the animation, and one in the Door for the logic. I'd highly recommend keeping these in one place (either the hand or the door, and just having that one call the other's Knock function from its Knock function). This is both for the sake of good practice and making the previous bullet point easier.
- Disable the hand and shadow sprites after the player has won. But with an adjustable buffer of like a fraction of a second so the last knock animation finished playing (so use Invoke)
- Instead of duplicating the door Animators like that, use an Animator Override Controlller. This allows you to use one base Animator for the flowchart/logic part, and each override controller allows you to change which animation happens in each state. In your case all the override values would be the same except for the appear/angry animation. Think of it the same way a prefab works but for having variants on the same Animator.
